### PR TITLE
Patches

### PR DESCRIPTION
**Pull request by maxime-poulain**

### CODE REVIEW

**Comment by arter97** [comment]
My patch is incompatible with Android 5.0 with Clang spitting out errors.
Have you confirmed it builds fine?

**Comment by maxime-poulain** [comment]
I guess, your build errors are due to the neon-vfpv4 that clang binary doesnt support, therefore a substitution to mfpu=neon is required.

Well, however, the clang binary from aosp master branch does support vfpv4 flag.

What clang compiler are you building with ?

File: core/clang/arm.mk [hunk 1]
```diff
@@ -27,6 +27,6 @@ CLANG_CONFIG_arm_UNKNOWN_CFLAGS := \
 define subst-clang-incompatible-arm-flags
   $(subst -march=armv5te,-march=armv5t,\
   $(subst -march=armv5e,-march=armv5,\
-  $(subst -mcpu=cortex-a15,-march=armv7-a,\
```
Comment (maxime-poulain) [inline core/clang/arm.mk:30]: This is removed because clang supports now this flag

File: core/clang/arm.mk [hunk 2]
```diff
@@ -27,6 +27,6 @@ CLANG_CONFIG_arm_UNKNOWN_CFLAGS := \
...
+  $(subst -mfpu=neon-vfpv4,-mfpu=neon,\
```
Comment (maxime-poulain) [inline core/clang/arm.mk:30]: current aosp clang binary doesnt suppor this flag , however clang 3.6+ does.

**Comment by arter97** [comment]
Ah, yes. You've fixed it.
But why is "-mcpu=cortex-a15,-march=armv7-a" removed?

**Comment by Khaon** [comment]
Hm maybe better to rebase with this ? 

https://android.googlesource.com/platform/build/+/a24a9a8e6266829b39f8dd44a4e31a12b4a69df0

**Comment by temasek** [comment]
arter97

I leave the decision to you.

**Comment by maxime-poulain** [comment]
Since I could find an AOSP commit that remove the fallback froma15 to generic, I rebased my patch set to dissociate it from the mfpu enhancement

**Comment by arter97** [comment]
So current cm-12.1's clang fully supports that, or do we need to update clang?

**Comment by maxime-poulain** [comment]
No need to update prebulit clang, current 3.5 snapshot from aosp supports mcpu=cortex-a15 but do not support neon-vfpv4 flags.

Well it is not a issue as clang automatically set hasVfp4 to true when the specified cpu is a15 !

**Comment by arter97** [comment]
Merged :)

**Pull request merged**